Framework/DataHandling/inc/MantidDataHandling/DefaultEventLoader.h

```diff
@@ -114,35 +118,34 @@ void DefaultEventLoader::makeMapToEventLists(
     // The index of eventVectors is a spectrum number so it is simply resized to
     // the maximum
     // possible spectrum number
-    eventid_max = maxSpecNo;
-    for (size_t i = 0; i < vectors.size(); ++i) {
-      vectors[i].resize(maxSpecNo + 1, nullptr);
-    }
+    m_maxEventID = maxSpecNo;
+    for (size_t i = 0; i < eventVectors.size(); ++i)
+      eventVectors[i].resize(maxSpecNo + 1, nullptr);
+
     for (size_t period = 0; period < m_ws.nPeriods(); ++period) {
       for (size_t i = 0; i < m_ws.getNumberHistograms(); ++i) {
         const auto &spec = m_ws.getSpectrum(i);
         getEventsFrom(m_ws.getSpectrum(i, period),
-                      vectors[period][spec.getSpectrumNo()]);
+                      eventVectors[period][spec.getSpectrumNo()]);
       }
     }
   } else {
-    // To avoid going out of range in the vector, this is the MAX INDEX that can
-    // go into it
-    eventid_max = static_cast<int32_t>(pixelID_to_wi_vector.size());
+    auto [minEventID, maxEventID] = getMinMaxDetID();
+    m_maxEventID = maxEventID;
 
     // Make an array where index = pixel ID
     // Set the value to NULL by default
-    for (size_t i = 0; i < vectors.size(); ++i) {
-      vectors[i].resize(eventid_max + 1, nullptr);
-    }
+    for (size_t i = 0; i < eventVectors.size(); ++i)
+      eventVectors[i].resize(m_maxEventID - minEventID + 1, nullptr);
 
-    for (size_t j = 0; j < pixelID_to_wi_vector.size(); j++) {
-      size_t wi = pixelID_to_wi_vector[j];
+    for (int32_t j = minEventID; j <= m_maxEventID; ++j) {
+      auto index = j + m_detIDtoIndexOffset;
+      size_t wi = m_detIDtoIndexVector[index];
```

Comment:
Does something guarantee that index is in the the range of `m_detIDtoIndexVector.size()`?

Reply:
Yes. The size of the vector is as large as the largest detector ID. The vector is created to improve access speed. See the corresponding method on `MatrixWorkspace` https://github.com/mantidproject/mantid/blob/master/Framework/API/src/MatrixWorkspace.cpp#L540

Reply:
Ah yes I'd forgotten about the dance this does with those vectors. Thanks.